src/Escape.elm

```diff
@@ -0,0 +1,55 @@
+module Escape exposing (SanitizeLevel(..), sanitize)
+
+import Regex exposing (Regex)
+
+
+type alias Replacement =
+    String
+
+
+type SanitizeLevel
+    = Namespace
+    | Element
+    | State
```

Comment:
I don't think it good idea to have "switch argument". It would be better to have separate functions for both performance and readability.